dataprofiler/profilers/histogram_utils.py

```diff
@@ -90,3 +366,61 @@ def _get_bin_edges(
         raise ValueError("`bins` must be 1d, when an array")
 
     return bin_edges, n_equal_bins
+
+
+def calculate_bins_from_profile(profile, bin_method):
+    """
+    Compute the bins used internally by `histogram`.
+
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+    :param bin_method: the method used to calculate bins based on the profile given
+    :type bin_method: string
+
+    :return: ideal number of bins for a particular histograom calulcation method
+    """
+    # parse the overloaded bins argument
+
+    rework_hist_bin_selectors = {
```

Comment:
instead of re init internal every time, maybe global in this file. Probably should privatize the variable too. Update name to suggest from profile in the name as oppose to rework.